File: Algorithms/p240_Search_a_2D_Matrix_II/p240_Search_a_2D_Matrix_II_2.py
class Solution(object):
    def searchMatrix(self, matrix, target):
        """
        :type matrix: List[List[int]]
        :type target: int
        :rtype: bool
        """
        if not matrix:
            return False
        m = len(matrix)
        n = len(matrix[0])
        i = 0
        j = n-1
        while i<m and j>=0:
            if matrix[i][j]<target:
                i += 1
            elif matrix[i][j]>target:
                j -= 1
            else:
                return True
        return False

File: Algorithms/p240_Search_a_2D_Matrix_II/p240_Search_a_2D_Matrix_II_2.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        if(matrix.size()==0) return 0;
        int m = matrix.size();
        int n = matrix[0].size();
        for(int i=0,j=n-1;i<m&&j>=0;){
            if(matrix[i][j]<target){
                ++i;
            }else if(matrix[i][j]>target){
                --j;
            }else{
                return 1;
            }
        }
        return 0;
    }
};
